### PR TITLE
Possible fix to flaky application_detail_view.feature test

### PR DESCRIPTION
**Pull request by kelbyhawn**
Adds wait time to the `application_detail_view.feature` test to potentially fix flakiness seen [here](https://cucumber-logs.s3.amazonaws.com/test/test/Safari_acquisition_products_pd_application_detail_view_output.html?versionId=nOjz6S9oaCXKkJr4FhB4ttcOf7f7Mh3p).

## Links
Jira ticket: [ACQ-1818](https://codedotorg.atlassian.net/browse/ACQ-1818)
Most recent Slack convo: [here](https://codedotorg.slack.com/archives/C04540KNGDQ/p1715617287413969)

## Testing story
Tested locally on Sauce Labs

### CODE REVIEW

**Review by hannahbergam — APPROVED** [review]
Makes sense to me! Were you able to visually confirm that it was looking for a button that didn't exist yet?

File: dashboard/test/ui/features/acquisition_products/pd/application_detail_view.feature [hunk 1]
```diff
@@ -9,6 +9,7 @@ Feature: Teacher Application Detail View
 
     Then I click selector "table#summary-csp-teachers ~ .btn:contains(View all applications)"
     Then I wait until element "#status-filter" is visible
+    And I wait for 3 seconds
```
Comment (bethanyaconnor) [inline dashboard/test/ui/features/acquisition_products/pd/application_detail_view.feature]: Could we instead wait for the button to be visible? (similar to line 14)

**Comment by kelbyhawn** [comment]
> Makes sense to me! Were you able to visually confirm that it was looking for a button that didn't exist yet?

[This is](https://app.saucelabs.com/tests/82715e23a3e14a5c919fea6ad4f2bb9e#32) the original failing test, but it kept passing for me locally. I did confirm that it waits 3 seconds once the table loads though.

File: dashboard/test/ui/features/acquisition_products/pd/application_detail_view.feature (hunk 1 above)
Comment (kelbyhawn) [reply dashboard/test/ui/features/acquisition_products/pd/application_detail_view.feature]: Yeah! I was going off of the flaky Facebook icon share test, but this feels more robust.

Comment (kelbyhawn) [reply dashboard/test/ui/features/acquisition_products/pd/application_detail_view.feature]: Updated, and here's the [passing video](https://app.saucelabs.com/tests/05543f85bda04b32976c6bc7dddb9ae0#58).

**Review by hannahbergam — COMMENTED** [review]
Thanks for updating this!

**Review by bethanyaconnor — APPROVED** [review]
Thanks for jumping on this so quickly!

**Pull request merged**